scripts/build_base_energy_totals.py

```diff
@@ -0,0 +1,382 @@
+import py7zr
+import requests
+from helpers import sets_path_to_root, three_2_two_digits_country
+
+pd.options.mode.chained_assignment = None
+
+
+def calc_sector(sector):
+    for country in countries:
+        # print(country, sector)
+        df_co = df_yr[df_yr.country == country]
+
+        if sector != "navigation":
+            df_sector = df_co.loc[
+                df["Commodity - Transaction"].str.lower().str.contains(sector)
+            ]
+            # assert df_yr[df_yr["Commodity - Transaction"].str.contains(sector)]["Unit"].unique() == 'Metric tons,  thousand', "Not all quantities have the expected unit: {}".format(expected_unit)
+        else:
+            df_sector = df_co.loc[
+                (df["Commodity - Transaction"].str.lower().str.contains(sector))
+                | (
+                    df["Commodity - Transaction"]
+                    .str.lower()
+                    .str.contains("marine bunkers")
+                )
+            ]
+
+        if df_sector.empty:
+            pass
+        else:
+            index_mass = df_sector.loc[
+                df_sector["Unit"] == "Metric tons,  thousand"
+            ].index
+            df_sector.loc[index_mass, "Quantity_TWh"] = df_sector.loc[index_mass].apply(
+                lambda x: x["Quantity"] * fuels_conv_toTWh[x["Commodity"]], axis=1
+            )
+
+            index_energy = df_sector[
+                df_sector["Unit"] == "Kilowatt-hours, million"
+            ].index
+            df_sector.loc[index_energy, "Quantity_TWh"] = df_sector.loc[
+                index_energy
+            ].apply(lambda x: x["Quantity"] / 1e3, axis=1)
+
+            index_energy_TJ = df_sector[df_sector["Unit"] == "Terajoules"].index
+            df_sector.loc[index_energy_TJ, "Quantity_TWh"] = df_sector.loc[
+                index_energy_TJ
+            ].apply(lambda x: x["Quantity"] / 3600, axis=1)
+
+            index_volume = df_sector[
+                df_sector["Unit"] == "Cubic metres, thousand"
+            ].index
+            df_sector.loc[index_volume, "Quantity_TWh"] = df_sector.loc[
+                index_volume
+            ].apply(lambda x: x["Quantity"] * fuels_conv_toTWh[x["Commodity"]], axis=1)
+
+            sectors_dfs[sector] = df_sector.copy()
+
+            if sector == "consumption by households":
+                energy_totals_base.at[country, "electricity residential"] = (
+                    df_sector[
+                        (df_sector.Commodity == "Electricity")
+                        | df_sector.Commodity.isin(other_fuels)
+                    ]
+                    .Quantity_TWh.sum()
+                    .round(4)
+                )
+                energy_totals_base.at[country, "residential oil"] = (
+                    df_sector[df_sector.Commodity.isin(oil_fuels)]
+                    .Quantity_TWh.sum()
+                    .round(4)
+                )
+                energy_totals_base.at[country, "residential biomass"] = (
```

Comment:
Could be a loop

Reply:
Could be, but honestly will be tedious to debug. I would vote to keep it like this as long as we have only one data source (unsd).